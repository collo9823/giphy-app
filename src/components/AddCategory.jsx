import React from "react";
import { useState } from "react";
import PropTypes from 'prop-types'

export const AddCategory = ({ onNewValue })=>{

    const [inputValue, setInputValue] = useState('');

    //Del input recibe el onChange y lo pasa al setInputValue
    const onChangeInput=(event)=>{
        setInputValue(event.target.value);
    }

    //Del form recibe el inputValue del input, trimea y si es may a 1 lo envÃ­a al valor a onNewValue
        //que es la propiedad. Resetea el setInputValue
    const onSubmit=(event)=>{
        event.preventDefault()

        if(inputValue.trim().length<=1)return; 
        onNewValue(inputValue.trim())
        setInputValue('')
    }

    return(
        <form onSubmit={(event)=>onSubmit(event)} aria-label='form'>
            <input
                type='text'
                placeholder="Buscar GIFs"
                value={inputValue}
                onChange={onChangeInput}
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewValue: PropTypes.func.isRequired,
}