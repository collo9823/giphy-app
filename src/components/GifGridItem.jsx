import React from "react";
import PropTypes from 'prop-types';

export const GifGridItem=({ title, src, id })=>{
    return(
        <div className="card">
            <img src={src} alt={title} />
            <p> {title} </p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    src: PropTypes.string.isRequired
}