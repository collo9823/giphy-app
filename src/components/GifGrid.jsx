import React from "react";
import {GifGridItem} from './GifGridItem'
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from 'prop-types'


export const GifGrid=({cat})=>{

    //Extrae los states images e isLoading de useFetchGifs, pasandol√© como argumento lo recibido en la propiedad cat
    const {images, isLoading} = useFetchGifs(cat);

    return(
        <>
            <h3>{cat}</h3>

            {
                isLoading && (<h2>Cargando...</h2>) 
            }
            

            <div className="card-grid" >

                {
                    images.map((image)=>(
                        <GifGridItem 
                            key={ image.id }
                            {...image}
                        />
                    ))
                        
                }
            </div>
        
        </>
    )
}

GifGrid.porpTypes = {
    category: PropTypes.string.isRequired,
}