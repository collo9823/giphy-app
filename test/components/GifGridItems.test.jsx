import { render, screen } from "@testing-library/react";
import { GifGridItem } from "../../src/components/GifGridItem";

describe('Pruebas en <GifGridItem/>', ()=>{
    const title= 'Programing';
    const src = 'https://media3.giphy.com/media/qgQUggAC3Pfv687qPC/giphy.gif?cid=83efc29fkosirvby0d0dy978twd4mj8pbtadgysbnn0671vp&rid=giphy.gif&ct=g'

    test('matchear con el snapshot', ()=>{
        const {container} = render(<GifGridItem title={title} src={src}/> );

        expect (container).toMatchSnapshot()
    })

    // test('Debe mostrar la imagen con el url y el alt indicado', ()=>{
    //     render( <GifGridItem title={title} src={src}/>);
    //     // screen.debug()
    //     // expect(screen.getByRole('img').src).toBe(src);

    //     const {src, alt} = screen.getByRole('img');
    //     // expect(src).toBe(src);
    //     expect(alt).toBe(title);
    // })

    test('Debe mostrar el tÃ­tulo en el componente', ()=>{
        render( <GifGridItem title={title} src={src}/>);
        expect(screen.getByText(title)).toBeTruthy();
    })
});