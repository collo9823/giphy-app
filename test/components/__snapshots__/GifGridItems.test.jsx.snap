// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> matchear con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Programing"
      src="https://media3.giphy.com/media/qgQUggAC3Pfv687qPC/giphy.gif?cid=83efc29fkosirvby0d0dy978twd4mj8pbtadgysbnn0671vp&rid=giphy.gif&ct=g"
    />
    <p>
       
      Programing
       
    </p>
  </div>
</div>
`;
